### PR TITLE
Cleanup code using walker->Weight

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Moving some code cleanup out of #4157. Nothing here should change behavior. 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Code style update (formatting, renaming)

### Does this introduce a breaking change?

- No

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes/No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes/No. Documentation has been added (if appropriate)

### CODE REVIEW

**Review by prckent — APPROVED** [review]
Very good to see all this wild point use removal.

For a later PR, I noticed some remnants of the since removed release node QMC method in this part of the code that deals with population control. e.g. write_release_nodes_, releasednodeage, ncopy_rn. These can also be deleted.

**Comment by prckent** [comment]
Test this please

**Comment by prckent** [comment]
Whoever looks to merge this (not at ORNL), please check all the += and similar lines carefully.

File: src/QMCDrivers/WalkerControlBase.cpp [hunk 1]
```diff
@@ -366,15 +359,8 @@ int WalkerControlBase::sortWalkers(MCWalkerConfiguration& W)
...
-    it_end = good_rn.end();
-    int indy(0);
-    while (it != it_end)
-    {
-      good_w.push_back(std::move(*it));
-      ncopy_w.push_back(ncopy_rn[indy]);
-      it++, indy++;
-    }
+    good_w.insert(good_w.end(), std::make_move_iterator(good_rn.begin()), std::make_move_iterator(good_rn.end()));
+    ncopy_w.insert(ncopy_w.end(), ncopy_rn.begin(), ncopy_rn.end());
```
Comment (ye-luo) [inline src/QMCDrivers/WalkerControlBase.cpp:363]: Assuming good_rn and ncopy_rn are equally sized. This change is valid.

**Review by ye-luo — APPROVED** [review]
All the changed lines under QMCDrivers belongs to legacy drivers. It reminds me of the importance of deleting legacy CPU drivers.

**Pull request merged**